### PR TITLE
MNT DOC Clear docs build from all warnings

### PR DESCRIPTION
**Pull request by TamaraAtanasoska**
## Description
This PR clears the doc build from all warnings: 
- [x]  WARNING: Underline too long
- [x]  WARNING: py:class reference target not found 
- [x]  WARNING: autosummary: stub file not found
- [x]  WARNING: image file not readable
- [x]  WARNING: undefined label
- [x]  WARNING: Inline interpreted text or phrase reference start-string without end-string.

Because of a conflict between the `napoleon` and `numpydoc` sphinx extensions which prompted the "`py:class reference target not found`"([more here](https://github.com/sphinx-doc/sphinx/issues/7391)), I removed the napoleon extension af9d1f968884732733874f10365368ff6993a230 and rewrote parts of the documentation in numpydoc style in f812ed4b75473b61ad1a2fe37c67c41ea232de6f.

This `numpydoc `class members suppression line in 275a70083b14938c025022f2aebd1ad58d5288c4 solves the `"autosummary: stub file not found"` warning, without changing the existing look of the documentation.

[Reference]( https://github.com/sphinx-gallery/sphinx-gallery/issues/120) for the solution for the`"image file not readable"` warning in 7e971320274bd8a28d90f8613e4b2cd24d2ff17f.

If there are still some warnings on another system, let me know! This makes a clean build for me completely. 

@adrinjalali 

## Tests

- [x] no new tests required
- [ ] new tests added
- [ ] existing tests adjusted

## Documentation

- [ ] no documentation changes needed
- [ ] user guide added or updated
- [x] API docs added or updated
- [ ] example notebook added or updated

### CODE REVIEW

File: fairlearn/reductions/_exponentiated_gradient/exponentiated_gradient.py [hunk 1]
```diff
@@ -32,8 +32,6 @@ class ExponentiatedGradient(BaseEstimator, MetaEstimatorMixin):
     The exponentiated gradient algorithm is described in detail by
     :footcite:t:`agarwal2018reductions`.
 
-    Read more in the :ref:`User Guide <exponentiated_gradient>`.
-
```
Comment (adrinjalali) [inline fairlearn/reductions/_exponentiated_gradient/exponentiated_gradient.py:36]: O_o don't we need to replace this reference with something that works?

File: docs/conf.py [hunk 2]
```diff
@@ -59,7 +59,6 @@
     "sphinx.ext.intersphinx",
     "sphinx.ext.linkcode",
     "sphinx.ext.mathjax",
-    "sphinx.ext.napoleon",
```
Comment (adrinjalali) [inline docs/conf.py:62]: ping @riedgar-ms re this

Comment (riedgar-ms) [reply docs/conf.py:62]: Something I did _four years_ ago??? :-)

No objection from me so long as the docs still work.

File: examples/plot_intersectional_bias.py [hunk 3]
```diff
@@ -186,7 +186,7 @@
 # ~~~~~~~~~~
 #
 #   **"For a star, the location of the mean referred to the location of a singular real object,
```
Comment (riedgar-ms) [inline examples/plot_intersectional_bias.py:188]: Should this be 'For a start' ?

**Review by riedgar-ms — APPROVED** [review]

File: fairlearn/reductions/_exponentiated_gradient/exponentiated_gradient.py (hunk 1 above)
Comment (TamaraAtanasoska) [reply fairlearn/reductions/_exponentiated_gradient/exponentiated_gradient.py:36]: I couldn't find what to replace it with concept wise after looking through the user guide, so I thought it got maybe outdated. If you have an idea what to link there I'd gladly do it :)

File: examples/plot_intersectional_bias.py (hunk 3 above)
Comment (TamaraAtanasoska) [reply examples/plot_intersectional_bias.py]: oops, thanks for catching that! fixed [f79d65e](https://github.com/fairlearn/fairlearn/pull/1378/commits/f79d65e250492a6cb70d6c0df7892defe7a7b7c3)

File: fairlearn/reductions/_exponentiated_gradient/exponentiated_gradient.py (hunk 1 above)
Comment (TamaraAtanasoska) [reply fairlearn/reductions/_exponentiated_gradient/exponentiated_gradient.py:36]: @adrinjalali maybe this one could be an option? https://fairlearn.org/v0.10/auto_examples/plot_mitigation_pipeline.html ?

Comment (adrinjalali) [reply fairlearn/reductions/_exponentiated_gradient/exponentiated_gradient.py:36]: Yeah it sucks that we still don't have a good doc for that. The example seems not bad.

Comment (TamaraAtanasoska) [reply fairlearn/reductions/_exponentiated_gradient/exponentiated_gradient.py]: I made a not for myself to create a better example in the future when there is time for it, and I added a link to the `plot_mitigation_pipeline` in [c79cc68](https://github.com/fairlearn/fairlearn/pull/1378/commits/c79cc68a043ec748b095b4793262d93f95f92d52).

**Review by romanlutz — APPROVED** [review]
I love these fixes!

**Comment by TamaraAtanasoska** [comment]
@adrinjalali the build finally passed with multiple re-triggers 🎉

File: fairlearn/reductions/_exponentiated_gradient/exponentiated_gradient.py [hunk 4]
```diff
@@ -32,6 +32,9 @@ class ExponentiatedGradient(BaseEstimator, MetaEstimatorMixin):
     The exponentiated gradient algorithm is described in detail by
     :footcite:t:`agarwal2018reductions`.
 
+    Read more in the \
+        `User Guide <https://fairlearn.org/main/auto_examples/plot_mitigation_pipeline.html>`_.
```
Comment (adrinjalali) [inline fairlearn/reductions/_exponentiated_gradient/exponentiated_gradient.py:36]: we should never reference the website, it should be referencing something that sphinx can resolve, example:

https://github.com/scikit-learn/scikit-learn/issues/26927

**Pull request merged**